test/perf/benchmarks/assets/compiler.test.ps1

```diff
@@ -0,0 +1,2686 @@
+        {
+            $packageDependenciesParams['Distribution']=$DebDistro
+        }
+
+        # Setup package dependencies
+        $Dependencies = @(Get-PackageDependencies @packageDependenciesParams)
+
+        $Arguments = Get-FpmArguments `
+            -Name $Name `
+            -Version $packageVersion `
+            -Iteration $Iteration `
+            -Description $Description `
+            -Type $Type `
+            -Dependencies $Dependencies `
+            -AfterInstallScript $AfterScriptInfo.AfterInstallScript `
+            -AfterRemoveScript $AfterScriptInfo.AfterRemoveScript `
+            -Staging $Staging `
+            -Destination $Destination `
+            -ManGzipFile $ManGzipInfo.GzipFile `
+            -ManDestination $ManGzipInfo.ManFile `
+            -LinkInfo $Links `
+            -AppsFolder $AppsFolder `
+            -Distribution $DebDistro `
+            -ErrorAction Stop
+
+        # Build package
+        try {
+            if ($PSCmdlet.ShouldProcess("Create $type package")) {
+                Write-Log "Creating package with fpm..."
+                $Output = Start-NativeExecution { fpm $Arguments }
+            }
+        } finally {
+            if ($Environment.IsMacOS) {
+                Write-Log "Starting Cleanup for mac packaging..."
+                if ($PSCmdlet.ShouldProcess("Cleanup macOS launcher"))
+                {
+                    Clear-MacOSLauncher
+                }
+
+                # this is continuation of a fpm hack for a weird bug
+                if (Test-Path $hack_dest) {
+                    Write-Warning "Move $hack_dest to $symlink_dest (fpm utime bug)"
+                    Start-NativeExecution -sb ([ScriptBlock]::Create("$sudo mv $hack_dest $symlink_dest")) -VerboseOutputOnError
+                }
+            }
+            if ($AfterScriptInfo.AfterInstallScript) {
+                Remove-Item -ErrorAction 'silentlycontinue' $AfterScriptInfo.AfterInstallScript -Force
+            }
+            if ($AfterScriptInfo.AfterRemoveScript) {
+                Remove-Item -ErrorAction 'silentlycontinue' $AfterScriptInfo.AfterRemoveScript -Force
+            }
+            Remove-Item -Path $ManGzipInfo.GzipFile -Force -ErrorAction SilentlyContinue
+        }
+
+        # Magic to get path output
+        $createdPackage = Get-Item (Join-Path $CurrentLocation (($Output[-1] -split ":path=>")[-1] -replace '["{}]'))
+
+        if ($Environment.IsMacOS) {
+            if ($PSCmdlet.ShouldProcess("Add distribution information and Fix PackageName"))
+            {
+                $createdPackage = New-MacOsDistributionPackage -FpmPackage $createdPackage -IsPreview:$IsPreview
+            }
+        }
+
+        if (Test-Path $createdPackage)
+        {
+            Write-Verbose "Created package: $createdPackage" -Verbose
+            return $createdPackage
+        }
+        else
+        {
+            throw "Failed to create $createdPackage"
+        }
+    }
+}
```

Comment:
Using our build and packaging code as a test case 🤣

Reply:
Yeah, I simply chose some big-size functions from `build.psm1` and `packaging.psm1`.
Not sure if that's a good idea though, because the scripting pattern of those functions are more or less the same -- a lot of if statements, a lot switch statements, and a lot command invocations.

Reply:
I think it is good real life scenario.